provisioner/src/lib.rs

```diff
@@ -338,6 +346,34 @@ impl MyProvisioner {
         })
     }
 
+    /// Send a request to the auth service with new subscription items that should be added to
+    /// the subscription of the [Claim] subject.
+    pub async fn add_subscription_items(
+        &self,
+        jwt: &str,
+        subscription_item: NewSubscriptionItem,
+    ) -> Result<(), Error> {
+        let response = self
+            .auth_client
+            .post(format!("{}users/subscription/items", self.auth_uri))
+            .bearer_auth(jwt)
+            .json(&subscription_item)
+            .send()
+            .await
+            .map_err(|err| {
+                error!(error = %err, "failed to connect to auth service");
+                Error::Plain("failed to connect to auth service".to_string())
+            })?;
+
+        match response.status() {
+            StatusCode::OK => Ok(()),
+            status_code => {
+                error!(status_code = %status_code, "failed to update subscription");
+                Err(Error::Plain("failed to update subscription".to_string()))
```

Comment:
That the deployment failed and that it was due to a failure to update the subscription will be visible to the user, but it will not be very useful. The most likely cause for this error will be that the JWT expired, but it should be very rare. We could consider having some suggestions to retry the command, and if it doesn't work reach out to support? :thinking:

Reply:
The JWT layer should return a 499 if that is the case, actually, so I should be able to match on that here and return a more appropriate message. I'll look into that.

Reply:
Addressed in https://github.com/shuttle-hq/shuttle/pull/1478/commits/4a500aedb1689b86dcdcddfab6b7ec5fb46024b9 and [e508559](https://github.com/shuttle-hq/shuttle/pull/1478/commits/e50855924b7dda0c957b2bd657953e026b95fd53).